File: BE/app/schemas/event.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class EventCreateRequest(BaseModel):
    camera_id: str
    detected_at: datetime
    anomaly_type: str
    confidence: float
    description: str | None = None


class EventStatusUpdateRequest(BaseModel):
    status: str


class EventItem(BaseModel):
    id: int
    camera_id: str
    ai_event_id: str | None = None
    detected_at: datetime
    ended_at: datetime | None = None
    duration_sec: float | None = None
    anomaly_type: str
    confidence: float
    description: str | None = None
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class EventDetail(EventItem):
    description: str | None
    video_url: str | None = None
    thumbnail_urls: list[str] = []


class EventListResponse(BaseModel):
    items: list[EventItem]
    total: int
    page: int
    size: int

File: fe/src/api/event.ts
import client from "./client";

export interface EventItem {
  id: number;
  camera_id: string;
  detected_at: string;
  anomaly_type: string;
  confidence: number;
  status: "UNREVIEWED" | "REVIEWED" | "FALSE_POSITIVE";
  created_at: string;
}

export interface EventDetail extends EventItem {
  description: string | null;
  video_url: string | null;
}

export interface EventListResponse {
  items: EventItem[];
  total: number;
  page: number;
  size: number;
}

export async function listEvents(params?: {
  page?: number;
  size?: number;
  status?: string;
  camera_id?: string;
}): Promise<EventListResponse> {
  const res = await client.get("/api/events", { params });
  return res.data.data;
}

export async function getEvent(id: number): Promise<EventDetail> {
  const res = await client.get(`/api/events/${id}`);
  return res.data.data;
}

export async function updateEventStatus(
  id: number,
  status: "REVIEWED" | "FALSE_POSITIVE"
): Promise<EventItem> {
  const res = await client.patch(`/api/events/${id}/status`, { status });
  return res.data.data;
}
